src/psyclone/psyir/transformations/omp_task_trans.py

```diff
@@ -0,0 +1,158 @@
+''' This module provides the OMPTaskTrans transformation.'''
+from psyclone.errors import GenerationError
+from psyclone.domain.common.transformations import KernelModuleInlineTrans
+from psyclone.psyGen import Kern
+from psyclone.psyir.transformations.fold_conditional_return_expressions_trans \
+        import FoldConditionalReturnExpressionsTrans
+from psyclone.psyir.transformations.parallel_loop_trans import\
+    ParallelLoopTrans
+from psyclone.psyir.nodes import CodeBlock, Call
+from psyclone.psyir.nodes import DynamicOMPTaskDirective
+from psyclone.psyir.transformations.inline_trans import InlineTrans
+from psyclone.psyir.transformations.transformation_error import \
+        TransformationError
+
+
+class OMPTaskTrans(ParallelLoopTrans):
+    ''' Apply an OpenMP Task Transformation to a Loop. The Loop must
+    be within an OpenMP Serial region (Single or Master) at codegen time.
+    '''
+
+    def __str__(self):
+        return "Adds an 'OMP TASK' directive to a statement"
+
+    @property
+    def name(self):
+        '''
+        :returns: the name of this transformation.
+        :rtype: str
+        '''
+        return "OMPTaskTrans"
+
+    def validate(self, node, options=None):
+        '''
+        Validity checks for input arguments.
+
+        :param node: the Loop node to validate.
+        :type node: :py:class:`psyclone.psyir.nodes.Loop`
+        :param options: a dictionary with options for transformations.
+        :type options: dict of string:values or None
+        '''
+        # Disallow CodeBlocks inside the region
+        if len(node.walk(CodeBlock)) > 0:
+            raise GenerationError(
+                "OMPTaskTransformation cannot be applied to a region "
+                "containing a code block")
+        super().validate(node, options)
+
+    def _directive(self, children, collapse=None):
+        '''
+        Creates the type of directive needed for this sub-class of
+        transformation.
+
+        :param children: list of Nodes that will be the children of \
+                         the created directive.
+        :type children: list of :py:class:`psyclone.psyir.nodes.Node`
+        :param collapse: A required parameter from parent class. Must
+                         never be set for TaskTrans (is None).
+        :type collapse: None.
+
+        :raises TransformationError: if the collapse attribute is set.
+
+        :returns: The directive created for the OpenMP Task Directive
+        :rtype: :py:class:`psyclone.psyGen.DynamicOMPTaskDirective`
+
+        '''
+        if collapse is not None:
+            raise TransformationError("Collapse attribute should not be set "
+                                      "for OMPTaskTrans")
+
+        _directive = DynamicOMPTaskDirective(children=children)
+        return _directive
+
+    def _inline_kernels(self, node):
+        '''
+        Searches the PsyIR tree inside the directive and inlines any kern
+        objects found.
```

Comment:
Please also specify that it first module-inlines them and folds any conditional return expressions.

Reply:
Fixed, and added what `node` is.